### ISSUE TITLE
[Package Issue Report]: bun

### ISSUE THREAD

**Issue by theurgi** [jetpack-io/devbox#1489]
**Are you requesting a new package or reporting an issue on an existing one?**

Reporting an issue with an existing package.

**If reporting an issue on an existing one, what is the name of the package?**
Please use the name as it appears in https://www.nixhub.io 

`bun`

**What changes are you requesting?**
Summarize your request. If it's for an existing package that is not
working as expected, provide a concise explanation of what is going wrong.

`bun` seems to install as expected:

```sh
❯ devbox add bun                                  

Installing package: bun@latest.

[1/1] bun@latest
[1/1] bun@latest: Success

❯ bun --version
1.0.2

❯ which bun   
# <project-path>/.devbox/virtenv/.wrappers/bin/bun
```

however, when I try to run `bun` or `bun --help`, it just hangs.

**Comment by Lagoja** [comment]
Hey @theurgi I tried to reproduce this issue with bun@1.0.2, but it seems to be running correctly on macOS without hanging. Can you share more information about the system you are testing this on?

**Comment by theurgi** [comment]
Sure! Some very strange behavior indeed. I haven't had issues with any other Devbox packages. Devbox is amazing by the way, I absolutely love it, it's my favorite new thing! 😊

System info:

```
macOS 10.15.7
devbox-0.5.11
nix (Nix) 2.17.0
/bin/zsh
```
I know macOS 10.15.7 is deprecated but newer versions don't support my hardware, thanks Apple. 🙃 It's okay, Linux is way cooler anyway. And if Devbox integrates Home Manager... dream system. 😍

## Reproduction

I've since made some changes on my system, so I'll recreate the setup and see if I encounter the same issue...

I originally had `bun` installed globally with `nix-env`:

```bash
❯ nix-env -q
bun-1.0.2

❯ which bun
/Users/user/.bun/bin/bun

❯ cat ~/.config/zsh/.zshrc
# ...
export BUN_INSTALL="$HOME/.bun"
export PATH="$BUN_INSTALL/bin:$PATH"
# bun completions
[ -s "$BUN_INSTALL/_bun" ] && source "$BUN_INSTALL/_bun"
# ...
```

I also wanted to create an isolated environment with Devbox:

```bash
~
❯ mkdir foo && cd foo

~/foo
❯ devbox init

~/foo
❯ devbox add bun
Ensuring nixpkgs registry is downloaded.
Downloaded 'github:NixOS/nixpkgs/8b5ab8341e33322e5b66fb46ce23d724050f6606' to '/nix/store/dwikaw0zm70m2jlv1ngjz2xf9j213jqm-source' (hash 'sha256-cJR9AFHmt816cW/C9necLJyOg/gsnkvEeFAfxgeM1hc=').
Ensuring nixpkgs registry is downloaded: Success

Installing package: bun@latest.

[1/1] bun@latest
[1/1] bun@latest: Success

~/foo
❯ devbox shell
Starting a devbox shell...

~/foo
❯ which bun
/Users/user/.bun/bin/bun # still the global version? 🤔
```

I thought maybe the global version was interfering and I really don't need a global version so...

```bash
❯ nix-env -e bun
uninstalling 'bun-1.0.2'
building '/nix/store/02lbcizi8442y58fcr03aj69vwp6ds8d-user-environment.drv'...

❯ nix-collect-garbage -d
removing old generations of profile /Users/user/.local/state/nix/profiles/profile ...

❯ cat ~/.config/zsh/.zshrc
# ...
#export BUN_INSTALL="$HOME/.bun"
#export PATH="$BUN_INSTALL/bin:$PATH"
# bun completions
#[ -s "$BUN_INSTALL/_bun" ] && source "$BUN_INSTALL/_bun"
# ...

> source ~/.config/zsh/.zshrc

~/foo
❯ which bun
/Users/user/.bun/bin/bun # still the global version? 🤔

> devbox rm bun # hangs ...

~/foo
❯ exit && cd # exit devbox shell

~
❯ which bun
/Users/user/.bun/bin/bun # wut??? very strange

~
> rm -rf .bun # this should do the trick

~
❯ which bun
bun not found # okay at least i'm not going insane 😅
```

Let's try again now that the global `bun` is gone.

```bash
~
> rm -rf foo && mkdir foo && cd foo

~/foo
❯ devbox init


~/foo
❯ devbox add bun
Ensuring nixpkgs registry is downloaded.
Downloaded 'github:NixOS/nixpkgs/8b5ab8341e33322e5b66fb46ce23d724050f6606' to '/nix/store/dwikaw0zm70m2jlv1ngjz2xf9j213jqm-source' (hash 'sha256-cJR9AFHmt816cW/C9necLJyOg/gsnkvEeFAfxgeM1hc=').
Ensuring nixpkgs registry is downloaded: Success

Installing package: bun@latest.

[1/1] bun@latest
[1/1] bun@latest: Success

~/foo
❯ devbox shell
Starting a devbox shell...

~/foo
❯ which bun
/Users/user/foo/.devbox/virtenv/.wrappers/bin/bun # Great!

~/foo
❯ bun
# blinking cursor ad infinitum... 
# same with `bun --help`. 😞

~/foo
❯ bun --version
1.0.2 # this works? ¯\_(ツ)_/¯


~/foo
❯ cat devbox.lock
{
  "lockfile_version": "1",
  "packages": {
    "bun@latest": {
      "last_modified": "2023-09-19T14:14:51Z",
      "resolved": "github:NixOS/nixpkgs/8b5ab8341e33322e5b66fb46ce23d724050f6606#bun",
      "source": "devbox-search",
      "version": "1.0.2"
    }
  }
}
```

**Comment by theurgi** [comment]
Spending a little more time with this and the issue seems to be between Nix, my system (macOS), and particularly `nixpkgs.bun`. I've not had this issue with any other Nix package.

Nix was installed on my system using the official and recommended multi-user installation script.

The strange thing is that, during the course of testing this by installing and uninstalling `bun` globally with `nix-env`, sometimes it worked while other times I experienced the same inability to run `bun` as I previously described.

In any case, this does not seem to be an issue with Devbox and it's not strictly blocking for me. I have no issues installing and running `bun` using the official script:

```sh
curl -fsSL https://bun.sh/install | bash
```

Just no isolated `bun` Devbox shell for now. 😞

Thanks for your help and consideration @Lagoja !
